src/store/targetDb.js

```diff
@@ -0,0 +1,17 @@
+import {List} from 'immutable'
+import {INIT_DB} from './db'
+
+export const UPDATE_DB = 'UPDATE_DB'
+
+const initialState = List([])
+
+export default function (state = initialState, action) {
+  switch (action.type) {
+    case INIT_DB:
+      return action.db
+    case UPDATE_DB:
+      return action.target_db
```

Comment:
Where is 'action.target_db coming from — I looked, but am not sure how it's defined. I'm sure I'm just missing some sort of logical shortcut.

Reply:
Not sure it's the best name but I'm calling 'target_db' the db with all of our changes for migration. So initially, we want to set it equal to 'db' which is the actual db as it is. I reused the INIT_DB action for that.  Then, once you make changes in the GUI, we will dispatch the UPDATE_DB action (should probably be UPDATE_TARGET_DB)  which will update the target_db with the changes

Reply:
Makes sense in a haze. But I'm just not sure at what point / where we're adding the key '.target_db' to the action. i.e., Where does this thing '.target_db' come from?

Maybe it's coming from here "INIT_DB" somehow? I'm sure this is really elementary.

I just have trouble following logic when I can't "see" where a value is defined.

Reply:
Oh yeah I see - I just haven't written the action / thunk yet. I'll write it in later